File: fc-chat/demo/part3/ch3/05-1-message-system-add-channel/src/main/java/net/prostars/messagesystem/auth/MessageUserDetailsService.java
package net.prostars.messagesystem.auth;

import net.prostars.messagesystem.entity.UserEntity;
import net.prostars.messagesystem.repository.UserRepository;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.security.core.userdetails.UserDetails;
import org.springframework.security.core.userdetails.UserDetailsService;
import org.springframework.security.core.userdetails.UsernameNotFoundException;
import org.springframework.stereotype.Service;

@Service
@SuppressWarnings("unused")
public class MessageUserDetailsService implements UserDetailsService {

  private static final Logger log = LoggerFactory.getLogger(MessageUserDetailsService.class);
  private final UserRepository userRepository;

  public MessageUserDetailsService(UserRepository userRepository) {
    this.userRepository = userRepository;
  }

  @Override
  public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
    UserEntity userEntity =
        userRepository
            .findByUsername(username)
            .orElseThrow(
                () -> {
                  log.info("User not found: {}", username);
                  return new UsernameNotFoundException("");
                });

    return new MessageUserDetails(
        userEntity.getUserId(), userEntity.getUsername(), userEntity.getPassword());
  }
}

File: fc-chat/demo/part3/with-push/src/main/kotlin/com/yscorp/withpush/messagesystem/auth/MessageUserDetailsService.kt
package com.yscorp.withpush.messagesystem.auth

import com.yscorp.withpush.messagesystem.entity.UserEntity
import com.yscorp.withpush.messagesystem.repository.UserRepository
import org.slf4j.Logger
import org.slf4j.LoggerFactory
import org.springframework.security.core.userdetails.UserDetails
import org.springframework.security.core.userdetails.UserDetailsService
import org.springframework.security.core.userdetails.UsernameNotFoundException
import org.springframework.stereotype.Service

@Service
@Suppress("unused")
class MessageUserDetailsService(userRepository: UserRepository) : UserDetailsService {
    private val userRepository: UserRepository = userRepository

    @Throws(UsernameNotFoundException::class)
    override fun loadUserByUsername(username: String): UserDetails {
        val userEntity: UserEntity =
            userRepository
                .findByUsername(username)
                ?: throw UsernameNotFoundException("")

        return MessageUserDetails(
            userEntity.userId, userEntity.username, userEntity.password
        )
    }

    companion object {
        private val log: Logger = LoggerFactory.getLogger(MessageUserDetailsService::class.java)
    }
}
